src/reuse/spdx.py

```diff
@@ -27,17 +28,17 @@ def run(args, project: Project, out=sys.stdout) -> int:
     """Print the project's bill of materials."""
     if args.file:
         out = args.file.open("w", encoding="UTF-8")
-        suffixes = args.file.suffixes
-        if suffixes[-1] == ".spdx" or (
-            len(suffixes) >= 2
-            and (
-                suffixes[-2] == ".spdx"
-                and suffixes[-1] in {".rdf", ".json", ".xml", ".yaml", ".yml"}
+        spdxfile = [
+            pattern
+            for pattern in _IGNORE_SPDX_PATTERNS
+            if pattern.match(args.file.name)
+        ]
+        if not spdxfile:
+            _LOGGER.warning(
+                _("'{path}' does not match a known SPDX file pattern").format(
+                    path=out.name
+                )
```

Comment:
I think this message is too abstract. Maybe something like: `"'{path}' does not end with .spdx or .spdx.ext"`, but I'm not a fan of that either.

Maybe provide a URL describing SPDX document file extensions or something?

Reply:
Sure, attempted it in another commit

Reply:
LGTM!